app/src/apps/commander/Editor.tsx

```diff
@@ -0,0 +1,189 @@
+                            <button
+                                className={`${tool === "remove" ? "underline" : ""} my-1`}
+                                onClick={() => setTool("remove")}
+                            >
+                                Remove
+                            </button>
+                        </li>
+                    </ul>
+                </div>
+            </div>
+        </div>
+    );
+
+    async function performAction(
+        _unit: typeof Unit.$inferType | null,
+        x: number,
+        y: number,
+    ) {
+        console.log("performAction", _unit, x, y);
+        console.log("tool", tool);
+        console.log("inventory", inventory);
+        if (tool === "place" && inventory) {
+            return placeFromInventory(x, y).then(() => setWait(false));
+        }
+
+        if (tool === "remove") {
+            return removeUnit(x, y).then(() => setWait(false));
+        }
+    }
+
+    async function removeUnit(x: number, y: number) {
+        if (!zkLogin.address) return;
+        if (!game) return;
+        setWait(true);
+
+        const tx = new Transaction();
+        tx.moveCall({
+            target: `${packageId}::commander::remove_unit`,
+            arguments: [
+                tx.object(game.id),
+                tx.pure.u16(x),
+                tx.pure.u16(y),
+            ],
+        });
+
+        await client.signAndExecuteTransaction({
+            signer: await flow.getKeypair({ network: "testnet" }),
+            requestType: "WaitForLocalExecution",
+            transaction: tx as any,
+        });
+
+        refetch();
+    }
+
+    async function placeFromInventory(x: number, y: number) {
+        if (!zkLogin.address) return;
+        if (!game) return;
+        setWait(true);
+
+        let unit;
+        const tx = new Transaction();
+        const team = tx.moveCall({
+            target: `${packageId}::commander::team_neutral`,
+        });
+
+        if (inventory == "soldier") {
+            unit = tx.moveCall({ target: `${packageId}::unit::soldier` });
+        }
+
+        if (inventory == "sniper") {
+            unit = tx.moveCall({ target: `${packageId}::unit::sniper` });
+        }
+
+        if (inventory == "barricade") {
+            unit = tx.moveCall({ target: `${packageId}::unit::barricade` });
+        }
+
+        if (inventory == "heavy") {
+            unit = tx.moveCall({ target: `${packageId}::unit::heavy` });
+        }
+
+        if (!unit) return console.log("unit not found");
+
+        tx.moveCall({
+            target: `${packageId}::commander::place_unit`,
+            arguments: [
+                tx.object(game.id),
+                tx.pure.u16(x),
+                tx.pure.u16(y),
+                unit,
+                team,
+            ],
+        });
+
+        await client.signAndExecuteTransaction({
+            signer: await flow.getKeypair({ network: "testnet" }),
+            requestType: "WaitForLocalExecution",
```

Comment:
I think this one is deprecated (and instead expects a `await client.waitForTransaction({ digest: res.digest})`

Reply:
Odd. No deprecation warning emitted, and if we were to update it, we should change the rest of the codebase.